File: backend/src/main/java/com/rajat/quickpick/model/CartItem.java
package com.rajat.quickpick.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CartItem {

    private String menuItemId;
    private String menuItemName;
    private String menuItemImage;
    private int quantity;
    private double unitPrice;
    private double totalPrice;
    private boolean isAvailable;
    private boolean isVeg;
}

File: frontend/shared/src/commonMain/kotlin/org/rajat/quickpick/domain/modal/cart/CartItem.kt
package org.rajat.quickpick.domain.modal.cart

import kotlinx.serialization.Serializable

@Serializable
data class CartItem(
    val menuItemId: String? = null,
    val menuItemName: String? = null,
    val menuItemImage: String? = null,
    val quantity: Int = 0,
    val unitPrice: Double = 0.0,
    val totalPrice: Double = 0.0,
    val isAvailable: Boolean = true,
    val isVeg: Boolean = false
)

